Dockerfile

```diff
@@ -21,13 +21,18 @@ RUN pecl channel-update pecl.php.net \
     && docker-php-ext-enable \
         apcu
 
+COPY docker/conf.d /usr/local/etc/php/conf.d
+
 RUN apt-get autoremove --purge -y && apt-get clean
 
-COPY docker/conf.d /usr/local/etc/php/conf.d
+RUN php -r "copy('http://getcomposer.org/installer', 'composer-setup.php');"
+RUN php composer-setup.php --install-dir=/usr/local/bin --filename=composer
+RUN php -r "unlink('composer-setup.php');"
 
 COPY . /app
 
 WORKDIR /app
 
-CMD ["php", "/app/build.php"]
+RUN composer install
```

Comment:
Running `composer install` after copying `/app` means if `composer.json` is changed (`git checkout 1.0.0; git checkout 1.0.1`) the source tree would have changed invalidating the build cache for `COPY . /app` causing all subsequent steps to re-execute. since `composer.lock` only ever exists inside the image